statistics/handle/handle.go

```diff
@@ -114,6 +114,286 @@ type Handle struct {
 	sysProcTracker sessionctx.SysProcTracker
 	// serverIDGetter is used to get server ID for generating auto analyze ID.
 	serverIDGetter func() uint64
+	// tableLocked used to store locked tables
+	tableLocked []int64
+}
+
+// GetTableLockedAndClear for unit test only
+func (h *Handle) GetTableLockedAndClear() []int64 {
+	tableLocked := h.tableLocked
+	h.tableLocked = make([]int64, 0)
+	return tableLocked
+}
+
+// LoadLockedTables load locked tables from store
+func (h *Handle) LoadLockedTables() error {
+	h.mu.Lock()
+	defer h.mu.Unlock()
+	ctx := kv.WithInternalSourceType(context.Background(), kv.InternalTxnStats)
+	rows, _, err := h.execRestrictedSQL(ctx, "SELECT table_id from mysql.stats_table_locked ")
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	for _, row := range rows {
+		tableID := row.GetInt64(0)
+		h.tableLocked = append(h.tableLocked, tableID)
+	}
+	// update SessionStatsCollector
+	{
+		prev := h.listHead
+		prev.Lock()
+		for curr := prev.next; curr != nil; curr = curr.next {
+			curr.Lock()
+			curr.tableLocked = h.tableLocked
+			curr.Unlock()
+		}
+		prev.Unlock()
+	}
+	// update SessionIndexUsageCollector
+	{
+		prev := h.idxUsageListHead
+		prev.Lock()
+		for curr := prev.next; curr != nil; curr = curr.next {
+			curr.Lock()
+			curr.tableLocked = h.tableLocked
+			curr.Unlock()
+		}
+		prev.Unlock()
+	}
```

Comment:
Do we also need to lock `SessionIndexUsageCollector`? It only tracks index usage and doesn't relate to statistics.

Reply:
OK, I will remvoe it.